### PR TITLE
Background Utility Classes

### PR DESCRIPTION
**Pull request by hynes-dialpad**
# Description
Update background color utility classes to make them immutable and reflective of the new utility class naming approach. Also add in background gradients for use with our new label work:

image

# Checklist
 - [x] Have you asked the contributors if a similar effort is already in process or has been solved?
 - [x] You've reviewed the [contribution guidelines](https://github.com/dialpad/dialtone/blob/staging/.github/CONTRIBUTING.md)?
 - [x] Your pull request uses `staging` as its base branch. (All PRs using `production` as its base branch will be declined).
 - [x] All `gulp` scripts successfully compile.
 - [x] Any affected documentation has been updated, removed, or extended.
 - [x] Provide a description what is being changed, extended, or removed.
 - [x] Link to any issue(s) this request is resolving.
 - [x] Request a review from Joshua Hynes, Ted Goas, or David Becher.

### CODE REVIEW

File: docs/utilities/visual/background-gradients.html [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+{% highlight html linenos %}
+<div class="d-bgg-blue-light d-bgp-sm">…</div>
+<div class="d-bgg-blue-light">…</div>
+<div class="d-bgg-blue-light d-bgp-lg">…</div>
+<div class="d-bgg-blue-light d-bgp-xl">…</div>
+{% endhighlight %}
+        </div>
+    </div>
+</section>
+<section class="d-stack16">
```
Comment (tedgoas-dialpad) [inline docs/utilities/visual/background-gradients.html:116]: What's the use case for Opacity? We're already providing a lot of options to manipulate a gradient. Does this give folks enough rope to hang themselves with?

File: docs/utilities/visual/backgrounds.html [hunk 2]
```diff
@@ -11,89 +11,74 @@ <h2 class="dialtone-title__section">Fills</h2>
             <h3 class="dialtone-title__subsection">{{ color.color }}</h3>
```
Comment (tedgoas-dialpad) [inline docs/utilities/visual/backgrounds.html:11]: Some text contrast issues with the darker colors in this page:
Screen Shot 2020-10-07 at 3 27 27 PM

File: docs/utilities/visual/backgrounds.html [hunk 3]
```diff
@@ -1,6 +1,6 @@
 ---
 layout: page
-title: Backgrounds
+title: Background Colors
 description: Dialtone provides both fill and gradient background classes. Most components have a background color set. These are to be used to override a component or for one-off designs.
 ---
 <section class="d-stack16">
```
Comment (tedgoas-dialpad) [inline docs/utilities/visual/backgrounds.html:6]: Why are the swatches on [backgrounds](https://deploy-preview-126--dialtone.netlify.app/utilities/visual/backgrounds/) two column, but [background gradients](https://deploy-preview-126--dialtone.netlify.app/utilities/visual/background-gradients/) is three? Can we make them match?

1 vote for three column :)

File: docs/utilities/visual/background-gradients.html [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    <header class="d-stack12">
+        <h2 class="dialtone-title__section">Opacity</h2>
+        <p class="dialtone-copy__normal">In addition to the percentage and direction controls above, Dialtone also provides the ability for you to control <strong>one</strong> background color's opacity using the <span class="code-example--inline">.d-bgo[#]</span> classes.</p>
+    </header>
+    <div class="dialtone-example" data-example="yes" data-code="yes">
+        <div class="dialtone-example__example">
+            <div class="dialtone-color-grid__group">
```
Comment (tedgoas-dialpad) [inline docs/utilities/visual/background-gradients.html:123]: If we keep this section, more contrast issues. Maybe align the text right?
Screen Shot 2020-10-07 at 3 37 31 PM

**Review by tedgoas-dialpad — CHANGES_REQUESTED** [review]
Couple bugs and a question about whether to offer opacity to background gradients.

File: docs/utilities/visual/background-gradients.html (hunk 1 above)
Comment (hynes-dialpad) [reply docs/utilities/visual/background-gradients.html:116]: I'd rather default to providing more flexibility than less. If we're seeing misuse we can always pull back IMO.

**Comment by hynes-dialpad** [comment]
@tedgoas-dialpad I've made the requested updates. If you could review and approve if you don't find anything else, that would be appreciated!

**Pull request merged**